riscv/decode.h

```diff
@@ -23,9 +23,12 @@
 
 typedef int64_t sreg_t;
 typedef uint64_t reg_t;
+typedef __int128 int128_t;
```

Comment:
Using `uint128_t` breaks compilation on 32-bit host systems.  I think it's OK to require 64-bit when using the vector extension, but we shouldn't break backwards compatibility when not using the vector extension.

One possible workaround is to do something like

    #ifdef __SIZEOF_INT128__
    typedef ...
    #endif

and in the definition of instructions that depend on int128, something like

    #ifdef __SIZEOF_INT128__
    // define the instruction
    #else
    abort_need_int128();
    #endif

Reply:
OK

Reply:
It is a little complicated issue. We uses double width type to handle widen integer instruction  or carry/borrow bit calculation.  128bit type is used to handle SEW =64 case.  For 32bit host, it means target can only support vector extension with ELAN = 32.

In my opinion,  for x86 host development environment, to drop i386 support is trend. Ubuntu have only provide x86_64 image since 17.10 and redhat also has remove it from RHEL 7.

It is reasonable to assume user's development is 64bit host.

Reply:
I will add 64bit check to configure.

Reply:
at last,  I add an AC_TYPE_CHECK for __int128_t but not 64bit host checkg.